pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -403,41 +396,47 @@ func (b *BridgeBindMechanism) discoverPodNetworkInterface() error {
 	}
 	b.podNicLink = link
 
-	// get IP address
 	addrList, err := b.handler.AddrList(b.podNicLink, netlink.FAMILY_V4)
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to get an ip address for %s", b.podInterfaceName)
 		return err
 	}
 	if len(addrList) == 0 {
-		b.dhcpConfig.IPAMDisabled = true
 		b.ipamEnabled = false
 	} else {
-		b.dhcpConfig.IP = addrList[0]
-		b.dhcpConfig.IPAMDisabled = false
 		b.ip = addrList[0]
 		b.ipamEnabled = true
 	}
 
-	if len(b.dhcpConfig.MAC) == 0 {
-		// Get interface MAC address
-		b.dhcpConfig.MAC = b.podNicLink.Attrs().HardwareAddr
-	}
-
 	if b.podNicLink.Attrs().MTU < 0 || b.podNicLink.Attrs().MTU > 65535 {
 		return fmt.Errorf("MTU value out of range ")
 	}
 
-	// Get interface MTU
-	b.dhcpConfig.Mtu = uint16(b.podNicLink.Attrs().MTU)
+	b.dhcpConfig = b.generateDhcpConfig()
```

Comment:
I"m not sure how does this pass compilation. Didn't you remove `dhcpConfig` from `BridgeBindMechanism`?